### PR TITLE
feat: make error objects serializable in checkout module

### LINKED ISSUE #3263: [depends on 3262] Make error objects serializable for checkout module
See master ticket: #3010

### PR DESCRIPTION
**Pull request by znikola**
closes #3263

### CODE REVIEW

File: projects/core/src/cart/store/effects/cart-entry.effect.ts [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
-import * as fromActions from './../actions';
+import { makeHttpErrorSerializable } from '../../../util/serialization-utils';
 import { CartEntryConnector } from '../../connectors/entry/cart-entry.connector';
+import * as fromActions from './../actions';
 
 @Injectable()
 export class CartEntryEffects {
   @Effect()
-  addEntry$: Observable<any> = this.actions$.pipe(
+  addEntry$: Observable<Action> = this.actions$.pipe(
```
Comment (bgambocjaviniar) [inline projects/core/src/cart/store/effects/cart-entry.effect.ts]: Put it to more specific fromActions.success | fail

File: projects/core/src/cart/store/effects/cart-entry.effect.ts [hunk 2]
```diff
@@ -21,14 +22,16 @@ export class CartEntryEffects {
...
+          catchError(error =>
+            of(new fromActions.AddEntryFail(makeHttpErrorSerializable(error)))
+          )
         )
     )
   );
 
   @Effect()
-  removeEntry$: Observable<any> = this.actions$.pipe(
+  removeEntry$: Observable<Action> = this.actions$.pipe(
```
Comment (bgambocjaviniar) [inline projects/core/src/cart/store/effects/cart-entry.effect.ts]: Put it to more specific fromActions.success | fail

File: projects/core/src/cart/store/effects/cart-entry.effect.ts [hunk 3]
```diff
@@ -38,13 +41,17 @@ export class CartEntryEffects {
...
+              new fromActions.RemoveEntryFail(makeHttpErrorSerializable(error))
+            )
+          )
         )
     )
   );
 
   @Effect()
-  updateEntry$: Observable<any> = this.actions$.pipe(
+  updateEntry$: Observable<Action> = this.actions$.pipe(
```
Comment (bgambocjaviniar) [inline projects/core/src/cart/store/effects/cart-entry.effect.ts]: Put it to more specific fromActions.success | fail

File: projects/core/src/checkout/store/effects/address-verification.effect.ts [hunk 4]
```diff
@@ -18,10 +16,10 @@ export class AddressVerificationEffect {
     map((action: any) => action.payload),
```
Comment (bgambocjaviniar) [inline projects/core/src/checkout/store/effects/address-verification.effect.ts]: maybe add a type here if possible

File: projects/core/src/checkout/store/effects/checkout.effect.ts [hunk 5]
```diff
@@ -157,7 +180,11 @@ export class CheckoutEffects {
             new fromActions.CreatePaymentDetailsSuccess(details),
           ]),
           catchError(error =>
-            of(new fromActions.CreatePaymentDetailsFail(error))
+            of(
+              new fromActions.CreatePaymentDetailsFail(
+                makeHttpErrorSerializable(error)
```
Comment (bgambocjaviniar) [inline projects/core/src/checkout/store/effects/checkout.effect.ts]: this is not catching the current unserializable error in payment form in checkout
- create new payment will throw the error, but this doesn't catch it

File: projects/core/src/checkout/store/effects/address-verification.effect.ts (hunk 4 above)
Comment (bgambocjaviniar) [inline projects/core/src/checkout/store/effects/address-verification.effect.ts]: maybe add a type here

File: projects/core/src/checkout/store/effects/checkout.effect.ts (hunk 5 above)
Comment (znikola) [reply projects/core/src/checkout/store/effects/checkout.effect.ts]: fixed

File: projects/core/src/checkout/store/effects/address-verification.effect.ts (hunk 4 above)
Comment (znikola) [reply projects/core/src/checkout/store/effects/address-verification.effect.ts]: done by adding a type to `ofType` operator above. if you hover of `action` now you'll see a type.

**Pull request merged**